actions/notification_actions.jsx

```diff
@@ -68,17 +68,21 @@ export function sendDesktopNotification(post, msgProps) {
         const member = getMyChannelMember(state, post.channel_id);
         const isCrtReply = isCollapsedThreadsEnabled(state) && post.root_id !== '';
 
-        if (!member || isChannelMuted(member) || userStatus === UserStatuses.DND || userStatus === UserStatuses.OUT_OF_OFFICE) {
+        if (!member || (!persistent && isChannelMuted(member)) || userStatus === UserStatuses.DND || userStatus === UserStatuses.OUT_OF_OFFICE) {
             return;
         }
 
-        let notifyLevel = member?.notify_props?.desktop || NotificationLevels.DEFAULT;
+        // we are bypassing channel member's notify props in the case of persistent notifications
+        let notifyLevel = (!persistent && member?.notify_props?.desktop) || NotificationLevels.DEFAULT;
 
         if (notifyLevel === NotificationLevels.DEFAULT) {
             notifyLevel = user?.notify_props?.desktop || NotificationLevels.ALL;
         }
 
-        if (notifyLevel === NotificationLevels.NONE) {
+        if (persistent && mentions.indexOf(currentUserId) === -1) {
```

Comment:
3/5 This seems wrong to me. I would think that only people who are supposed to get a persistent notification would receive a websocket event for it.

Reply:
We could change it to send only to mentioned users, that's true.

cc @vish9812

Reply:
Yes we are changing this before merging, Vishal is working on it on the server side